easybuild/easyconfigs/g/GROMACS/GROMACS-2023.2-cpeGNU-22.12-HeFFTe-GPU.eb

```diff
@@ -0,0 +1,127 @@
+# Based on recipe from Rasmus Kronberg (CSC) 
+#
+easyblock = 'CMakeMake'
+
+name =          'GROMACS'
+version =       '2023.2'
+versionsuffix = '-HeFFTe-GPU' 
+
+homepage = 'http://www.gromacs.org'
+
+whatis = [
+    'Description: GROMACS (with multi-GPU acceleration), a versatile package to perform molecular dynamics, i.e. simulate the Newtonian equations of motion for systems with hundreds to millions of particles.',
+]
+
+description = """
+GROMACS is a versatile package to perform molecular dynamics, i.e. simulate the
+Newtonian equations of motion for systems with hundreds to millions of
+particles. This version enables GPU acceleration with PME decomposition feature 
+allowing distribution across multiple nodes and GPUs (supports ROCm via hipSYCL).
+
+Binaries included:
+* gmx_mpi: Multi-node MPI version with HeFFTe for a multi-GPU decomposition.
+
+The module sets the environment variables that would otherwise be set by running
+the script $EBROOTGROMACS/bin/GMXRC. So running this script snould not be
+}necessary in most cases.
+
+The module also defines the shell function gromacs-completion that can be used to
+turn bash command completion on for GROMACS. Note that unloading the module will
+not turn this off, so it may have consequences for things you do later on or
+conflict with the command completion of a different version of GROMACS.
+
+"""
+
+toolchain = {'name': 'cpeGNU', 'version': '22.12'}
+toolchainopts = {'opt': True, 'usempi': True, 'verbose': False, 'pic': True, 'openmp': True}
+
+source_urls = ['https://ftp.gromacs.org/pub/gromacs/']
+sources =     [SOURCELOWER_TAR_GZ]
+
+patches = [
+    'gromacs_cmake_manage_sycl.patch',
+]
+
+builddependencies = [
+    ('buildtools', '%(toolchain_version)s', '', True), 
+]
+
+dependencies = [
+    ('hipSYCL', '0.9.4', '-rocm-5.2.3'), 
+    ('HeFFTe', '2.3.0', '-rocm-5.2.3'),
+    ('Boost', '1.81.0'),
+    ('cray-fftw', EXTERNAL_MODULE),
+    ('cray-python', EXTERNAL_MODULE),
+    ('cray-libsci/22.12.1.1', EXTERNAL_MODULE),
+    ('rocm/5.2.3', EXTERNAL_MODULE),
+]
+
+# These options are taken from CSC installation developed by Rasmus 
+
+configopts =  '-D CMAKE_BUILD_TYPE=Release '
+configopts += '-D CMAKE_C_COMPILER=${ROCM_PATH}/llvm/bin/clang '
+configopts += '-D CMAKE_CXX_COMPILER=${ROCM_PATH}/llvm/bin/clang++ '
+configopts += '-D CMAKE_PREFIX_PATH=${CRAY_MPICH_PREFIX} '
+configopts += '-D CMAKE_CXX_FLAGS="--gcc-toolchain=/opt/cray/pe/gcc/11.2.0/snos/" '
+configopts += '-D GMX_GPU=SYCL '
+configopts += '-D GMX_MPI=ON '
+configopts += '-D GMX_OPENMP=ON '
+configopts += '-D GMX_SYCL_HIPSYCL=ON '
+configopts += '-D GMX_USE_HEFFTE=ON '
+configopts += '-D GMX_GPU_FFT_LIBRARY=rocFFT '
+configopts += '-D GMX_BUILD_OWN_FFTW=ON '
```

Comment:
Small note: `cray-fftw/3.3.10.1` works fine in our experience. Not that it will make a lot of difference for GPU runs.